### PR TITLE
Update WinAppSDK versions to latest public builds

### PR DESCRIPTION
**Pull request by jonthysell**
## Description

This PR updates the default versions of `Microsoft.WindowsAppSDK` used to 1.4.240211001 when `<UseWinUI3>true</UseWinUI3>` is specified, and 1.5.240124002-experimental2 when `<UseExperimentalWinUI3>true</UseExperimentalWinUI3>` is specified.

This PR also updates all of the `packages.lock json` files to the defaults.

### Type of Change

- Bug fix (non-breaking change which fixes an issue)

### Why
To keep up with the latest public versions of WinAppSDK.

### What
See above.

## Screenshots
N/A

## Testing
Ran both PR and new Secure PR checks to make sure that everything still builds and works.

## Changelog
Should this change be included in the release notes: yes

Updated WindowsAppSDK dependency to 1.4.240211001

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/react-native-windows/pull/12747)

### CODE REVIEW

File: vnext/PropertySheets/WinUI.props [hunk 1]
```diff
@@ -7,8 +7,8 @@
       For local testing of internal versions, modify the WinUI3Version, and comment out the addition nuget source in NuGet.Config
     -->
     <!-- This value is also used by the CLI, see /packages/@react-native-windows/generate-windows -->
-    <WinUI3Version Condition="'$(WinUI3Version)'=='' AND '$(UseExperimentalWinUI3)'=='true'">1.5.231202003-experimental1</WinUI3Version>
-    <WinUI3Version Condition="'$(WinUI3Version)'==''">1.4.230913002</WinUI3Version>
+    <WinUI3Version Condition="'$(WinUI3Version)'=='' AND '$(UseExperimentalWinUI3)'=='true'">1.5.240205001-preview1</WinUI3Version>
```
Comment (acoates-ms) [inline vnext/PropertySheets/WinUI.props:10]: I think the preview build will not have the experimental features in it, so probably wont work?

Comment (jonthysell) [reply vnext/PropertySheets/WinUI.props:10]: Ah. Mostly just trying an experiment in getting the Secure PR to run.

**Review by TatianaKapos — APPROVED** [review]

**Pull request merged**